### ISSUE TITLE
NKAI doesn't move at all on Barren Lands map

### ISSUE THREAD

**Issue by DjWarmonger** [vcmi/vcmi#3599]
**Describe the bug**
AI just passes turns and all enemies lose at day 8.
Could be because the only way out of starting position in Barren Lands is through one-way portal.

**To Reproduce**

- Launch HoTA map "Barren Lands"
- Use `vcmigod` and capture closest town behind the portal
- Use autoskip and just pass turns

**Expected behavior**
AI would play normally and at least try to move outisde starting spot

**Actual behavior**
AI just passes turn and does nothing, losing without town after a week

**Did it work earlier?**
No idea

**Version**
1.4.5

**Comment by nullkiller** [comment]
Yes, one-way monoliths is the thing which is kind of problem. I do not know exact conditions when it is ok to pass through one-way monolith. Sometimes it can act as a trap for a weak hero. So it should be a main hero and either we should be able to go back or the thing on the other side should be very important.
